src/shaders/basic_shader.cpp

```diff
@@ -36,23 +39,61 @@ void BasicShader::draw(Mesh *mesh, SceneInfo &scene_info, glm::mat4 to_world) {
         set_uni("texture_map", 1); // ID of this texture=1
     }
 
-    /* SHADOW AND TRANSFORMATION MATRICES */
+    /* SHADOW MATRICES */
     // Send shadow uniforms and bind its texture.
     set_uni("light_matrix", ShaderManager::shadow.light_matrix);
     glActiveTexture(GL_TEXTURE0);
     glBindTexture(GL_TEXTURE_2D, ShaderManager::shadow.shadow_map_tex);
     set_uni("shadow_map", 0); // ID of this texture=0
 
-    // Send lighting uniforms.
-    set_uni("dir_light.direction", -scene_info.light_pos);
-    set_uni("dir_light.color", Color::WHITE.to_vec()); // white light
-    set_uni("dir_light.ambient_coeff", 0.2f);
+    /* LIGHTS */
+    // Send directional lights.
+    std::string uni_prefix = "dir_lights[";
```

Comment:
This string stuff is kinda 🤔 
Care to leave a comment explaining what's being done?